deltacat/catalog/model/catalog.py

```diff
@@ -213,108 +195,101 @@ def init(
     from factory methods like Catalog.default or Catalog.iceberg
-
     Otherwise, this function initializes the catalog by using the catalog implementation provided by `impl`.
 
     Args:
         name: name of catalog
         catalog: catalog instance to use, if provided
         impl: catalog module to initialize. Only used if `catalog` param not provided
-        ray_init_args: ray initialization args (used if ray must be initialized)
+        default:  Make this the default catalog if multiple catalogs are available.
+            ignored if this is the only catalog available, since it will always be the default catalog.
+        ray_init_args: ray initialization args (used only if ray not already initialized)
+        fail_if_exists: if True, raises KeyError if the catalog name already exists. Otherwise, overwrite catalog
     """
-    # Set namespace to default, or use value in kwargs or ray_init_args
-    if kwargs.get("namespace") is not None:
-        namespace = kwargs.get("namespace")
-        # ensure ray init arg set with this namespace
-        ray_init_args = ray_init_args if ray_init_args is not None else {}
-        ray_init_args["namespace"] = kwargs.get("namespace")
-    elif ray_init_args and "namespace" in ray_init_args:
-        namespace = ray_init_args["namespace"]
-    else:
-        namespace = _default_namespace
+    global all_catalogs
 
-    if catalog is None:
-        catalog = Catalog(impl, **kwargs)
-    elif catalog is not None and impl != DeltacatCatalog:
-        raise ValueError(
-            f"PutCatalog call provided both `impl` and `catalog` parameters. "
-            f"You may only provide one of these parameters"
-        )
+    # Initialize, if necessary
+    if not is_initialized():
+        # NOTE - since we are initializing with a single catalog, it will be set to the default
+        if not default:
+            logger.info(
+                f"Calling put_catalog with set_as_default=False, "
+                f"but still setting Catalog {catalog} as default since it is the only catalog."
+            )
+        init({name: catalog}, ray_init_args=ray_init_args)
+        return
 
-    if is_initialized(**{"namespace": namespace}):
+    # Fail if fail_if_exists and catalog already exists
+    if fail_if_exists:
+        catalog_already_exists = False
         try:
-            get_catalog(name, namespace=namespace)
+            get_catalog(name)
+            # Note - need to set state catalog_already_exists and throw ValueError later, or else it will be
+            # caught in the except block which is meant to catch the ValueError from get_catalog
+            catalog_already_exists = True
+        except ValueError:
+            pass
+        if catalog_already_exists:
             raise ValueError(
-                f"Catalog {name} already exists in namespace '{namespace}'."
+                f"Failed to put catalog {name} because it already exists and fail_if_exists=True"
```

Comment:
For both a sanity check of this error message and to (hopefully) draw attention of future devs who may change the `fail_if_exists` param name, can we change this to:
```suggestion
                f"Failed to put catalog {name} because it already exists and fail_if_exists={fail_if_exists}"
```